src/memtable.rs

```diff
@@ -181,6 +187,18 @@ impl<E: Message + Clone, W: EntryExt<E>> MemTable<E, W> {
         self.cut_entries_cache(first_index_to_add);
         self.cut_entries_index(first_index_to_add);
 
+        if let Some((queue, index)) = self.entries_index.back().map(|e| (e.queue, e.index)) {
+            if first_index_to_add > index + 1 {
+                assert_eq!(
+                    queue,
+                    LogQueue::Rewrite,
+                    "memtable {} has a hole",
+                    self.region_id
+                );
+                self.compact_to(index + 1);
```

Comment:
Would it be a corruption that may lose data?

Reply:
It could be a corruption. Do you have any suggestions?

Reply:
There could still be a corruption even if we do the index contiguous check when `self.entries_index` is empty. So, I prefer to omit the check.

Reply:
Maybe we can provide a public method to verify this from the upper level application

Reply:
The API is not suitable here, because we can't get `truncated_index` from raft engine.